src/R/Components/Impl/InteractiveWorkflow/Commands/SessionInformationCommand.cs

```diff
@@ -35,12 +39,28 @@ public sealed class SessionInformationCommand : IAsyncCommand {
             return CommandResult.Executed;
         }
 
-        private void OnBrokerChanged(object sender, EventArgs e) {
+        private void OnRSessionInitialized(object sender, EventArgs e) {
             if (_interactiveWorkflow.RSession.IsRemote) {
-                PrintBrokerInformationAsync(reportTelemetry: true).DoNotWait();
+                ReplInitComplete().ContinueWith(async (t) => await PrintBrokerInformationAsync(reportTelemetry: true)).DoNotWait();
```

Comment:
You can move RData into separate class, but you can't use new `Initialized` event for it, cause we have to execute initialization code in a specific order, and events have no determination on that.

In fact, this is fired in the middle of nowhere. On the one hand, there is `Connected` event already, which is raised when RHost loop is ready for input (and if you call any evaluation at that moment, it will be executed after all that initialization code). On the other hand, some complex functions (like rtvs:::install.packages) require RHost to get to the state when it can accept user input, which has nothing to do with `Initialized` event either. And this event has nothing to do with Interactive Window readiness that you need in this particular case (cause you don't really care about session - you need broker and console to print information).